assembly.py

```diff
@@ -471,6 +472,52 @@ def order_and_orient(inFasta, inReference, outFasta,
     #    os.unlink(trimmed)
     return 0
 
+def _call_order_and_orient_orig(kwargs):
+    _order_and_orient_orig(**kwargs)
+
+def order_and_orient(inFasta, inReference, outFasta,
+        outAlternateContigs=None, outReference=None,
+        breaklen=None, # aligner='nucmer', circular=False, trimmed_contigs=None,
+        maxgap=200, minmatch=10, mincluster=None,
+        min_pct_id=0.6, min_contig_len=200, min_pct_contig_aligned=0.3, n_genome_segments=0, threads=0):
+    ''' This step cleans up the de novo assembly with a known reference genome.
+        Uses MUMmer (nucmer or promer) to create a reference-based consensus
+        sequence of aligned contigs (with runs of N's in between the de novo
+        contigs).
+    '''
+
+    ref_segments_all = tuple(Bio.SeqIO.parse(inReference, 'fasta'))
+    n_genome_segments = n_genome_segments or len(ref_segments_all)
+    assert (len(ref_segments_all) % n_genome_segments) == 0
+    n_refs = int(len(ref_segments_all) / n_genome_segments)
+    log.info('n_genome_segments={} n_refs={}'.format(n_genome_segments, n_refs))
+
+    with util.file.tempfnames([ '.{}.ref.fasta'.format(ref_num) for ref_num in range(n_refs)]) as refs_fasta, \
+         util.file.tempfnames([ '.{}.scaffold.fasta'.format(ref_num) for ref_num in range(n_refs)]) as scaffolds_fasta, \
+         util.file.tempfnames([ '.{}.altcontig.fasta'.format(ref_num) for ref_num in range(n_refs)]) as alt_contigs_fasta:
+
+         for ref_num in range(n_refs):
+             Bio.SeqIO.write(ref_segments_all[ref_num*n_genome_segments : (ref_num+1)*n_genome_segments], refs_fasta[ref_num], 'fasta')
+
+         pool = multiprocessing.Pool(threads or None)
```

Comment:
`threading` might be a better choice here, especially if it's shelling out to other processes anyways

Reply:
you mean to use multiprocessing.dummy ?

it's not just shelling out to other processes, it's calling tools.mummer.MummerTool.scaffold_contigs_custom() which does work in Python.  My understanding was that Python's Global Interpreter Lock limits parallelism?

Reply:
What about `concurrent.futures.ProcessPoolExecutor`, which is used in a few other places already?

Reply:
That's not in Python 2.6, right?

Reply:
2.7

Reply:
It's not, but the 2.7 build of viral-ngs [depends](https://github.com/broadinstitute/viral-ngs/blob/master/requirements-py2.txt) on the [`futures` package](https://pythonhosted.org/futures/) which provides it.

Reply:
changed it to use concurrent.futures.ProcessPoolExecutor